spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/pre/PreDecorationFilter.java

```diff
@@ -149,6 +152,9 @@ else if (!xforwardedfor.contains(remoteAddr)) { // Prevent duplicates
 					}
 					ctx.addZuulRequestHeader("X-Forwarded-For", xforwardedfor);
 				}
+				if (this.properties.isAddHostHeader()) {
```

Comment:
That's an of course an option. 

For the current default tomcat configuration of spring-boot a combination of `X-Forwarded-Proto` and `Host` is required. A buggy implementation could ignore the `*-Proto` header if the `*-Host` header is missing so it might be better to have two identical headers - `Host` and `X-Forwarded-Host` - for best results.

But since HTTP proxy headers are not standardized one could probably make a point to make all headers configurable. But in this case, a customizer interface would probably the better call.